app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -35,12 +35,25 @@ class BloodPressureEntrySheet : BottomSheetActivity() {
   private val onDestroys = PublishSubject.create<Any>()
 
   companion object {
-    private const val KEY_PATIENT_UUID = "patientUuid"
+    private const val KEY_OPEN_AS = "openAs"
+    private const val KEY_UUID = "uuid"
     private const val EXTRA_WAS_BP_SAVED = "wasBpSaved"
 
-    fun intent(context: Context, patientUuid: UUID) =
-        Intent(context, BloodPressureEntrySheet::class.java)
-            .putExtra(KEY_PATIENT_UUID, patientUuid)!!
+    private enum class OpenAs {
+      NEW_BP, UPDATE_BP
+    }
+
+    fun intentForNewBp(context: Context, patientUuid: UUID): Intent {
+      return Intent(context, BloodPressureEntrySheet::class.java)
+          .putExtra(KEY_OPEN_AS, OpenAs.NEW_BP)
+          .putExtra(KEY_UUID, patientUuid)
+    }
+
+    fun intentForUpdateBp(context: Context, bloodPressureMeasurementUuid: UUID): Intent {
+      return Intent(context, BloodPressureEntrySheet::class.java)
+          .putExtra(KEY_OPEN_AS, OpenAs.UPDATE_BP)
+          .putExtra(KEY_UUID, bloodPressureMeasurementUuid)
```

Comment:
OMG the same key is being used for two different types of values. Let's not do this? 😱

Reply:
Vinay, Sanchita and I were talking that `OpenAs` can be converted into a sealed class to offer type safety in the controller.

```kotlin
sealed class OpenAs {
  data class New(val patientUuid: UUID) : OpenAs()
  data class Update(val bpUuid: UUID) : OpenAs()
}
```

Reply:
Yeah, I didn't like it either.

Can I use `@Parcelize` on a sealed class? That would solve a lot of problems.

Reply:
Done.